File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int tar) {
        for(int i=0;i<nums.size();++i){
            int idx = upper_bound(begin(nums),end(nums),tar-nums[i])-nums.begin();
            cout<<idx<<endl;
            if(idx-1<nums.size() and nums[idx-1]==tar-nums[i]){
                vector<int>v;
                v.push_back(i+1);
                v.push_back(idx);
                return v;
            }
        }
        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], tar: int) -> List[int]:
        low, high = 0,len(nums)-1
        while(low<high):
            if nums[low]+nums[high]==tar:
                return [low+1,high+1]
            if nums[low] + nums[high]>tar:
                high-=1
            else:
                low+=1
        return [0,0]